Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1639,38 +1759,39 @@ void GridTrackSizingAlgorithm::resolveIntrinsicTrackSizes(GridLayoutState& gridL
 
 void GridTrackSizingAlgorithm::resolveIntrinsicTrackSizesMasonry(GridLayoutState& gridLayoutState)
 {
-    if (m_strategy->isComputingSizeContainment()) {
+    if (m_strategy->isComputingSizeContainment() || !m_grid.hasGridItems()) {
         handleInfinityGrowthLimit();
         return;
     }
+    StdMap<SpanLength, MasonryMinMaxTrackSize> indefiniteItems;
+    StdMap<SpanLength, Vector<MasonryMinMaxTrackSizeWithGridSpan>> definiteItems;
+    Vector<MasonryMinMaxTrackSizeWithGridSpan> definiteItemsThatSpanFlexTrack;
```

Comment:
Nit: I feel like "That," in this name doesn't really add anything so this could probably be called `definiteItemsSpanningFlexTracks`

Reply:
That sounds a bit better; will update.